miscellaneous/miscellaneous.js

```diff
@@ -0,0 +1,150 @@
+        for(let i = 0; i<RPSScore.length; i++)
+        {
+            if(RPSScore[i].userID === userId) {
+                user = RPSScore[i];
+                break;
+            }
+        }
+        if(user === undefined)
+        {
+            user.userId = userId;
+            user.w = 0; //wins
+            user.l = 0; //loses
+            user.t = 0;//ties
+            user.r =0;//rock
+            user.p=0;//paper
+            user.s =0;//scissors
+        }
+        let cpuPick = RPS[Math.floor(Math.random()*RPS.length)];
+        let message = '';
+        //User and computer picked the same
+        if(cpuPick.toLowerCase() === userInput.toLowerCase())
+        {
+            user.t++;
+            message = 'You picked '+cpuPick.toLowerCase()+', cpu picked '+userInput.toLowerCase()+'. You tied. <@'+ userID +'>\n' +
+                ' wins ' + user.w +
+                '\nloses '+ user.l +
+                '\nties '+ user.t
+        }
+        else  if(userInput.toLowerCase() === 'rock')
+        {
+            if(cpuPick.toLowerCase() === 'paper'){
+                user.l++;
+                message = 'You picked rock, cpu picked paper. You lost. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+            else
+            {
+                user.w++;
+                message = 'You picked rock, cpu picked scissors. You won. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+        }
+        else  if(userInput.toLowerCase()=== 'paper')
+        {
+            if(cpuPick.toLowerCase() === 'scissors'){
+                user.l++;
+                message = 'You picked paper, cpu picked scissors. You lost. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+            else
+            {
+                user.w++;
+                message = 'You picked paper, cpu picked rock. You won. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+        }
+        else
+        {
+            if(cpuPick.toLowerCase() === 'rock'){
+                user.l++;
+                message = 'You picked paper, cpu picked paper. You lost. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+            else
+            {
+                user.w++;
+                message = 'You picked paper, cpu picked scissors. You won. <@'+ userID +'>\n' +
+                    ' wins ' + user.w +
+                    '\nloses '+ user.l +
+                    '\nties '+ user.t
+            }
+        }
+        let newUser = true;
```

Comment:
starting from this point I get lost on what's going on

Reply:
I imagine that it's recording the record of the player and figuring out if it's a new player or not.